pandas/core/internals/blocks.py

```diff
@@ -2104,7 +2105,10 @@ def is_view(self) -> bool:
 
     @property
     def array_values(self) -> ExtensionArray:
-        return NumpyExtensionArray(self.values)
+        if self.values.dtype.kind == 'T':
+            return NumpyStringArray(self.values)
+        else:
+            return NumpyExtensionArray(self.values)
```

Comment:
This is because NumpyExtensionArray inherit from `ObjectStringArrayMixin`, so unless I make this change there's no way to call the string method overrides in `NumpyStringArray`.